pages/learn/elective.tsx

```diff
@@ -0,0 +1,246 @@
+        </title>
+      </Head>
+
+      {/* Background */}
+      <div
+        className={cn(`fixed inset-0 bottom-auto -z-10 hidden h-screen
+          overflow-hidden sm:block`)}
+      >
+        <LandingBlobs className="inset-0" />
+      </div>
+
+      {/* Content */}
+      <PageHeader parentURL="/learn">
+        {t("title", { context: "student" })}
+      </PageHeader>
+      <ContentLayout
+        className={cn(`grow *:h-full *:!gap-6 sm:*:!grid
+          md:*:grid-cols-[5fr,7fr]`)}
+      >
+        <section className="flex-col gap-3 space-y-3 sm:flex">
+          {/* List */}
+          <div
+            className={cn(`grow sm:overflow-auto sm:rounded-xl
+              sm:bg-surface-bright md:h-0`)}
+          >
+            <List className="sm:!py-2">
+              {electiveSubjects.map((electiveSubject) => (
+                <ElectiveListItem
+                  key={electiveSubject.id}
+                  electiveSubject={electiveSubject}
+                  selected={radioSelected === electiveSubject.id}
+                  onRadioToggle={(value) => {
+                    if (value) setRadioSelected(electiveSubject.id);
+                  }}
+                  onClick={() => {}}
+                />
+              ))}
+            </List>
+          </div>
+
+          {/* Choose Button */}
+          <Actions
+            className={cn(`pointer-events-none sticky inset-0 bottom-20 top-auto
+              z-10 !-mt-6 !block bg-gradient-to-t from-surface-container p-4
+              pt-12 sm:static sm:!mt-0 sm:!flex sm:bg-none sm:p-0 sm:px-0`)}
+          >
+            <Button
+              appearance="filled"
+              icon={<MaterialIcon icon="done" />}
+              className="!pointer-events-auto"
+            >
+              {t("list.action.choose", { context: "initial" })}
+            </Button>
+          </Actions>
+        </section>
+
+        <div
+          className={cn(`flex flex-col gap-6 *:rounded-xl *:bg-surface-bright
+            md:pb-[3.25rem]`)}
+        >
+          {/* Details */}
+          <ElectiveDetailsCard className="hidden grow md:flex" />
+
+          {/* Trade */}
+          <TradesCard
+            className={cn(`mb-16 h-36 !rounded-b-none sm:m-0
+              sm:!rounded-b-xl`)}
+          />
+        </div>
+
+        <style jsx global>{`
+          body {
+            background-color: var(--surface-container);
+          }
+
+          .skc-root-layout {
+            display: flex;
+            flex-direction: column;
+            height: 100dvh;
+          }
+
+          @media only screen and (min-width: 600px) {
+            .skc-nav-bar::before {
+              background-color: transparent !important;
+            }
+
+            .skc-page-header__blobs {
+              display: none !important;
+            }
+          }
+        `}</style>
+      </ContentLayout>
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
```

Comment:
Is this function providing sample data? It seems to be missing it's classes and why is the class size a fixed number.

Reply:
It's providing **elecitve** class data, so that's why there is both a cap size and the current class size.

Reply:
This function provides props to the page—one prop just happens to be sample data while we wait for the API. Check out [Next.js’ documentation on `getServerSideProps`](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props).